sdk/python/kfp/components/pipeline_context.py

```diff
@@ -13,18 +13,14 @@
 # limitations under the License.
 """Definition for Pipeline."""
 
-import textwrap
 from typing import Callable, Optional
 
+from kfp.components import component_factory
+from kfp.components import graph_component
 from kfp.components import pipeline_task
 from kfp.components import tasks_group
 from kfp.components import utils
 
-# This handler is called whenever the @pipeline decorator is applied.
-# It can be used by command-line DSL compiler to inject code that runs for every
-# pipeline definition.
-pipeline_decorator_handler = None
-
 
 def pipeline(name: Optional[str] = None,
```

Comment:
Doesn't pipeline name fall back to a sanitized version of the pipeline function name? It's possible I'm misunderstanding somewhere else name is used. In the following example, I do not provide a pipeline name, yet there is one in IR:

```python
from kfp import dsl
from kfp import compiler
import yaml


@dsl.component
def nothing():
    pass


@dsl.pipeline()
def my_pipeline():
    nothing()


output = 'out.yaml'
compiler.Compiler().compile(pipeline_func=my_pipeline, package_path=output)

with open(output, 'r') as f:
    ir = yaml.safe_load(f.read())

assert ir['pipelineInfo']['name'] == 'my-pipeline'
```

If I am correct, my current thinking is that irrespective of the introduction of graph components we should consider implementing the `@dsl.pipeline` decorator similarly to the `@dsl.component` decorator and not requiring that it be called when used.

This is somewhat unrelated to your PR (other than the removal of those tests) and I think this can be implemented in a separate PR if we go this route.

Reply:
I thought we required users to provide a name explicitly because of it importance--being used for pipeline context. Maybe at some point our implementation added the fallback logic so that explicit name is no longer required. 

Other than the name requirement, I can't think of any reason that we want to enforce it to be called.